src/libstd/ffi/c_str.rs

```diff
@@ -1228,9 +1236,18 @@ impl CStr {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub unsafe fn from_ptr<'a>(ptr: *const c_char) -> &'a CStr {
-        let len = sys::strlen(ptr);
-        let ptr = ptr as *const u8;
-        CStr::from_bytes_with_nul_unchecked(slice::from_raw_parts(ptr, len as usize + 1))
+        // SAFETY: The caller has provided a pointer that:
+        //  - is a valid C string with a NUL terminator of size less than `isize::MAX`,
+        //  - will not change (nor the content) before the CStr is destroyed.
```

Comment:
I personally find this wording a little confusing, but maybe it's just me. Here's a suggestion:
```suggestion
        // SAFETY: The caller has provided a pointer that points to a valid C string with
        // a NUL terminator of size less than `isize::MAX`, whose content remain valid
        // and doesn't change for the lifetime of the returned `CStr`.
```